src/TwitterDB.js

```diff
@@ -0,0 +1,462 @@
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.get_users_from_dm_room = function(room_id){
+  log.info("TwitDB","Retrieving dm room: %s",room_id);
+  return new Promise((resolve,reject) =>{
+    this.db.get(
+      `
+      SELECT users
+      FROM dm_room
+      WHERE dm_room.room_id = $room_id;
+      `
+    ,{
+      $room_id: room_id
+    }
+    ,(err,row) =>{
+      if(err != null){
+        log.error("TwitDB","Error retrieving dm room: %s",err.Error);
+        reject(err);
+      }
+      if(row !== undefined){
+        resolve(row.users);
+      }
+      else {
+        resolve(null);
+      }
+    });
+  });
+}
+
+TwitterDB.prototype.add_dm_room = function(room_id,users){
+  log.info("TwitDB","Storing dm room %s",room_id);
+  this.db.run(
+    `
+    INSERT INTO dm_room VALUES ($room_id,$users);
+    `
+  ,{
+    $room_id: room_id,
+    $users: users
+  },
+  function (err) {
+    if(err){
+      log.error("TwitDB","Error storing dm room: %s",err);
+      return;
+    }
+    log.info("TwitDB","Stored dm room %s",room_id);
+  });
+}
+
+//Caches every user profile we grab from Twitter so as to not go over our limits.
+TwitterDB.prototype._create_profile_cache = function(){
+  this.db.run(
+    `
+    CREATE TABLE IF NOT EXISTS user_cache (
+    	id	INTEGER UNIQUE NOT NULL,
+      screenname TEXT NOT NULL,
+    	profile	TEXT NOT NULL,
+    	timestamp	INTEGER NOT NULL,
+    	PRIMARY KEY(id)
+    )
+    `,
+    function (err) {
+      if(err){
+        throw "Error creating 'user_cache': "+err;
+      }
+    }
+  );
+}
+
+//Keeps track of links between matrix users and their accounts
+TwitterDB.prototype._create_twitter_table = function(){
+  this.db.run(
+    `
+    CREATE TABLE IF NOT EXISTS twitter_account (
+      user_id	TEXT UNIQUE NOT NULL,
+      oauth_token TEXT,
+      oauth_secret	TEXT,
+      access_token TEXT,
+      access_token_secret	TEXT,
+      twitter_id	INTEGER,
+      PRIMARY KEY(user_id)
+    )
+    `,
+    function (err) {
+      if(err){
+        throw "Error creating 'twitter_account': "+err;
+      }
+    }
+  );
+}
+
+//Keeps track of links between matrix users and their accounts
+TwitterDB.prototype._create_timeline_table = function(){
+  this.db.run(
+    `
+    CREATE TABLE IF NOT EXISTS timeline_room (
+      user_id	TEXT UNIQUE NOT NULL,
+      room_id	TEXT NOT NULL,
+      PRIMARY KEY(user_id)
+    )
+    `,
+    function (err) {
+      if(err){
+        throw "Error creating 'twitter_account': "+err;
+        return;
+      }
+    }
+  );
+}
+
+TwitterDB.prototype._create_dm_room_table = function(){
+  this.db.run(
+    `
+    CREATE TABLE IF NOT EXISTS dm_room (
+    	room_id	TEXT NOT NULL,
+    	users	TEXT NOT NULL,
+    	PRIMARY KEY(room_id)
+    );
+    `,
+    function (err) {
+      if(err){
+        throw "Error creating 'dm_room': "+err;
+      }
+    }
+  );
+}
+
+
+TwitterDB.prototype.close = function() {
+  this.db.close();
+}
+
+module.exports = {
+  TwitterDB: TwitterDB
+}
```

Comment:
Your setters in general are missing return Promises - Some parts of the bridge *do care* if the data they tried to persist actually worked or not.